src/core/lib/config/core_configuration.cc

```diff
@@ -16,10 +16,13 @@
 
 #include "src/core/lib/config/core_configuration.h"
 
+#include <atomic>
+
 #include <grpc/support/log.h>
 
 namespace grpc_core {
 
+std::atomic<Mutex*> CoreConfiguration::builder_mu_;
```

Comment:
Why do we need this mutex now, when we didn't before?

If we are going to have a mutex here, do we still need atomics?  Could we just use `gpr_once_init()` to set a non-atomic global `Mutex*` variable, and then just use that mutex for the remaining data?

Can we add lock annotations, so that it's clear to both readers and to compiler checks which variables are being guarded by this mutex?

Can I have a sentence in this comment that is not in the form of a question? :)

Reply:
We used to allow multiple builders to run concurrently and race to provide the finished configuration: since that happens relatively infrequently, and the configuration is relatively fast to construct, it seemed like a good trade-off.

What changes here is that the service config code stores some of its configuration in static variables (specifically the indices of the configured service config slots).

We could eliminate this mutex if we stored those indices as part of the configuration, which I considered... however that would mean some additional indirections I believe along the data path to load the core configuration and chase a pointer to get the relevant index. I guess if we had some scheme to locate those indices on the channel that might alleviate it and re-simplify this code.

Reply:
I'm not sure I stated this strongly enough:

I would be strongly in favor of removing this mutex and finding some other way of storing those indices.